src/lib.rs

```diff
@@ -910,13 +921,13 @@ pub mod pallet {
 			Self::deposit_event(Event::NewDataFeed {
 				query_id,
 				feed_id,
-				query_data,
+				query_data: query_data.clone(),
```

Comment:
Probably better to simply capture the len() of query data into a variable before even emitted, to avoid cloning a potentially large vec.